test/lit/passes/remove-unused-brs_branch-hints-shrink.wast

```diff
@@ -0,0 +1,180 @@
+  ;; CHECK-NEXT:    (local.get $y)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $join-br_ifs-0 (param $x i32) (param $y i32)
+    ;; The hints once more match, but now are 0. We still propagate.
+    (block $out
+      (@metadata.code.branch_hint "\00")
+      (br_if $out
+        (local.get $x)
+      )
+      (@metadata.code.branch_hint "\00")
+      (br_if $out
+        (local.get $y)
+      )
+
+      (call $none)
+      (br_if $out
+        (local.get $y)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $join-br_ifs-no (type $0) (param $x i32) (param $y i32)
+  ;; CHECK-NEXT:  (block $out
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (i32.or
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (local.get $y)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:   (call $none)
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (local.get $y)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $join-br_ifs-no (param $x i32) (param $y i32)
+    ;; One is missing a hint, so we clear the hint.
+    (block $out
+      (@metadata.code.branch_hint "\00")
+      (br_if $out
+        (local.get $x)
+      )
+      (br_if $out
+        (local.get $y)
+      )
+
+      (call $none)
+      (br_if $out
+        (local.get $y)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $join-br_ifs-no-flip (type $0) (param $x i32) (param $y i32)
+  ;; CHECK-NEXT:  (block $out
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (i32.or
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (local.get $y)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:   (call $none)
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (local.get $y)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $join-br_ifs-no-flip (param $x i32) (param $y i32)
+    ;; The other one is missing the hint, so we clear the hint.
+    (block $out
+      (br_if $out
+        (local.get $x)
+      )
+      (@metadata.code.branch_hint "\00")
+      (br_if $out
+        (local.get $y)
+      )
+
+      (call $none)
+      (br_if $out
+        (local.get $y)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $join-br_ifs-mismatch (type $0) (param $x i32) (param $y i32)
+  ;; CHECK-NEXT:  (block $out
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (i32.or
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (local.get $y)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:   (call $none)
+  ;; CHECK-NEXT:   (br_if $out
+  ;; CHECK-NEXT:    (local.get $y)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $join-br_ifs-mismatch (param $x i32) (param $y i32)
+    ;; The hints do not match, so we clear the hint.
+    (block $out
+      (@metadata.code.branch_hint "\00")
+      (br_if $out
+        (local.get $x)
+      )
+      (@metadata.code.branch_hint "\01")
```

Comment:
If $x is unlikely but $y is likely, then `$x or $y` should still be likely, so it seems best to keep the likely hint in this case.